### DISCUSSION TITLE
what does: Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.1))" that exactly mean?

### DISCUSSION THREAD

**Discussion by jcompagner** [eclipse-tycho/tycho#1084] [Q&A] [answered]
Hi,

this is not directly related to tycho, but i know there are a lot that knows a lot about this stuff here (and i don't know where i should exactly report this for eclipse)

The new eclipse (i think WebTools.. not sure) has suddenly upgraded the apache xml resolver
it was: org.apache.xml.resolver_1.2.0.v201005080400.jar 

but now with the latest eclipse if you download for example 
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/2022-06/R/eclipse-jee-2022-06-R-win32-x86_64.zip

it is: org.apache.xml.resolver_1.2.0.v20220401-1849.jar

there are quite some changes in the manifest for that, including:

Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.1))" 

if i read that correctly, does that say that it requires java 1.1 ??

because now if i have my target platform all to the new 4.24/2022.06 and i validate my launch config..

![image](https://user-images.githubusercontent.com/3274268/176128452-5a605d80-5ad4-4ed6-a04b-89030f9768b8.png)

(my execution environment is java 17)

so is this a bug now in the new manifest? they entered something wrong there, or does the validate of the launch target report this wrong?

**Comment by laeubi** [comment]
> Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.1))"
> so is this a bug now in the new manifest?

Yes see http://docs.osgi.org/reference/eenames.html JavaSE-1.1 is not a valid EE

> if i read that correctly, does that say that it requires java 1.1

Don't know if it is useful these days, but if it really want to target Java 1.1, the lowest java version official supported is `J2SE-1.2` or `OSGi/Minimum-1.2` (**and not JavaSE-1.2** !)

**Reply by akurtakov** [reply]
J2SE-1.2 is for the BREE but for osgi.ee capability it has to be JavaSE . It's another question whether JavaSE-1.1 is supported version based on the the fact that 1.1 is not supposed for BREE,  it shouldn't be supported for osgi.ee capability.

**Reply by laeubi** [reply]
> J2SE-1.2 is for the BREE but for osgi.ee capability it has to be JavaSE

Do you have a link at hand for this?

> It's another question whether JavaSE-1.1 is supported version based on the the fact that 1.1 is not supposed for BREE, it shouldn't be supported for osgi.ee capability.

1.2 is the lowest java version an OSGi framework can run on, so it really makes no sense to declare 1.1 as you never could run such a bundle on java 1.1 :-)

**Reply by akurtakov** [reply]
> > J2SE-1.2 is for the BREE but for osgi.ee capability it has to be JavaSE
> 
> Do you have a link at hand for this?

Sure, it's your own link http://docs.osgi.org/reference/eenames.html#osgiee-capability . There is no J2SE valid value for osgi.ee . Note how BREE: J2SE-1.5 but osgi.ee JavaSE-1.5 .

**Reply by laeubi** [reply]
Thanks I must have missed that :+1:

**Comment by akurtakov** [comment]
It's definetely not a Tycho problem. I strongly recommend people to stop using Orbit artifacts or to work on Orbit so it stops overwriting original authors OSGi manifests.  I replied to the PDE issue on the topic here https://github.com/eclipse-pde/eclipse.pde/issues/179#issuecomment-1164200600 . See the discussion there - there are more than 1 issue in different places and different projects/plugins but Java 1.1 is not smth worth spending time on.

**Comment by jcompagner** [comment, marked as answer]
yeah i found it...

the supported list of profiles is in the OSGI plugin:

![image](https://user-images.githubusercontent.com/3274268/176134453-84a5e051-4e6a-4837-a34b-44cc254d0363.png)


so yes as far as i read if you want to target 1.1 they should have taken JRE-1.1 not JavaSE-1.1

I have to say for some jars we also do it, but we then just fork it..
Because if the original developers don't care about osgi then it is very tricky, i need to "osgify" the jar..
I guess thats why on maven we have for some jars so many different releases over different groups...

**Reply by akurtakov** [reply]
Yes, valid entries are specified in https://github.com/eclipse-equinox/equinox/blob/master/bundles/org.eclipse.osgi/JRE-1.1.profile#L21:
```
org.osgi.framework.system.capabilities = \
 osgi.ee; osgi.ee="JRE"; version:List<Version>="1.0, 1.1"
```
Best is to report to/fix in Orbit so it doesn't put invalid osgi.ee requirements .

**Reply by jcompagner** [reply]
https://bugs.eclipse.org/bugs/show_bug.cgi?id=580280

**Comment by Bananeweizen** [comment]
Not sure if it's a valid workaround for you, but I have just added the old version of xml.resolver to our target (and use it explicitly in launch configs) and have not run into trouble until now.
